browser/installer/windows/nsis/shared.nsh

```diff
@@ -1000,6 +1000,12 @@
   ; Remove protocol handler registry keys added by the MS shim
   DeleteRegKey HKLM "Software\Classes\Waterfox.URL"
   DeleteRegKey HKCU "Software\Classes\Waterfox.URL"
+
```

Comment:
The part "Remove protocol handler registry keys added by the MS shim" did reference Firefox until I changed it in #348 to be Waterfox. I'm not what the "MS shim" refers to but figured it should be changed.

On deleting the keys/values of Firefox-$AppUserModelID, I checked the installer code and it turns out that $AppUserModelID is a hash value calculated from the installation path.

Because of that, the name was unique and did not conflict with Firefox but I changed it in pull request #362 to keep the use of the name Waterfox consistent and because I found code that needed $AppRegName to be Waterfox to be consistent with the hard coded strings that I had to change to Waterfox instead of Firefox to avoid conflicts with Firefox.

Thus Waterfox 56.0.3 uses the [ClientID] of Waterfox-$AppUserModelID and doesn't know that previous versions used Firefox-$AppUserModelID so on upgrade, there would be two ClientIDs for Waterfox which causes two entries to appear in Set Default Programs for Waterfox so this patch will remove that key so one one Waterfox entry appears in Set Default Programs.

So in summery: This keys existence has no affect on Firefox, as the installation path would need to be the same for the $AppUserModelID parts to match and installing Waterfox in the same directory as an install of Firefox would cause many more problems then this.